app/iot/service.py

```diff
@@ -18,32 +18,54 @@ def connect(self) -> None:
     def disconnect(self) -> None:
         ...
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
+    async def send_message(self, message_type: MessageType, data: str) -> None:
         ...
 
 
 class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
+        attempts_left = 7
+        while device_id in self.devices:
+            print(
+                f"Device with id {device_id} already registered"
+                f"Gererating new one..."
+            )
+            device_id = generate_id()
+            attempts_left -= 1
+            if attempts_left == 0:
+                print("Unable to register device, trying again...")
+            print(f"Attempts left: {attempts_left}")
         self.devices[device_id] = device
         return device_id
 
-    def unregister_device(self, device_id: str) -> None:
-        self.devices[device_id].disconnect()
-        del self.devices[device_id]
+    async def unregister_device(self, device_id: str) -> None:
+        try:
+            self.devices[device_id].disconnect()
+            del self.devices[device_id]
+        except KeyError:
+            print(f"Device {device_id} not found")
 
-    def get_device(self, device_id: str) -> Device:
-        return self.devices[device_id]
 
-    def run_program(self, program: list[Message]) -> None:
+    async def get_device(self, device_id: str) -> Device:
+        try:
+            return self.devices.get(device_id)
+        except AttributeError:
+            print(f"Device {device_id} not found")
+
+
+    async def run_program(self, program: list[Message]) -> None:
         print("=====RUNNING PROGRAM======")
         for msg in program:
-            self.send_msg(msg)
+           await self.send_msg(msg)
         print("=====END OF PROGRAM======")
 
-    def send_msg(self, msg: Message) -> None:
-        self.devices[msg.device_id].send_message(msg.msg_type, msg.data)
+    async def send_msg(self, msg: Message) -> None:
+       try:
+           await self.devices.get(msg.device_id).send_message(msg.msg_type, msg.data)
+       except AttributeError:
+           print("Device not found")
```

Comment:
Code quality issue: It is a good practice to provide more information in error messages. In this case, it would be helpful to include the device_id in the error message.